flycheck.el

```diff
@@ -2414,7 +2452,14 @@ buffer manually.
               next-error-function
             :unset))
     (when flycheck-standard-error-navigation
-      (setq next-error-function #'flycheck-next-error-function)))
+      (setq next-error-function #'flycheck-next-error-function))
+
+    ;; This hook must be added globally since otherwise we cannot
+    ;; detect a change from a buffer where Flycheck is enabled to a
+    ;; buffer where Flycheck is not enabled, and therefore cannot
+    ;; notice that there has been any change when the user switches
+    ;; back to the buffer where Flycheck is enabled.
+    (add-hook 'buffer-list-update-hook #'flycheck-handle-buffer-switch))
```

Comment:
The hook is never removed, right? If not, we should make sure that flycheck is enabled as the first thing that flycheck-handle-buffer-switch does

Reply:
We can't do that. At the very least, we have to record that the user switched buffers; otherwise, it's not detected when the user switches from a Flycheck buffer to a non-Flycheck buffer and then back.

It would be better to make sure the hook is removed when it is no longer necessary, I think.

Reply:
OK, I'm onboard with that :)